src/function/gds/output_writer.cpp

```diff
@@ -80,7 +80,7 @@ static void setLength(ValueVector* vector, uint16_t length) {
     vector->setValue<uint16_t>(0, length);
 }
 
-void PathsOutputWriter::write(processor::FactorizedTable& fTable, nodeID_t dstNodeID) const {
+offset_t PathsOutputWriter::write(processor::FactorizedTable& fTable, nodeID_t dstNodeID) {
```

Comment:
I think you can push limit down all the way here too. That would be the most efficient. That is, you can push the limit to both here but you still have to implement some logic to stop the frontier morsel grabbing and calling vertex computes. That would be the most efficient but it's your call.

Reply:
I already pushed into output writer. Which push down are u referring here?